addons/web/static/src/js/views/list/list_renderer.js

```diff
@@ -847,23 +849,118 @@ var ListRenderer = BasicRenderer.extend({
      * @param {KeyboardEvent} ev
      */
     _onKeyDown: function (ev) {
-        if (!this.editable) {
-            switch (ev.which) {
-                case $.ui.keyCode.DOWN:
-                    $(ev.currentTarget).next().find('input').focus();
-                    ev.preventDefault();
-                    break;
-                case $.ui.keyCode.UP:
-                    $(ev.currentTarget).prev().find('input').focus();
-                    ev.preventDefault();
-                    break;
-                case $.ui.keyCode.ENTER:
-                    ev.preventDefault();
-                    var id = $(ev.currentTarget).data('id');
+        var findConnectedCell = function ($cell, direction, colIndex) {
+
+            var findConnected = function (element, selector) {
+                if (direction === 'prev') {
+                    return element.prev(selector);
+                } else {
+                    return element.next(selector);
+                }
+            }
+
+            var connectedRow = findConnected($cell.closest('tr'), 'tr');
+
+            if (!connectedRow.length) {
+                // Is there another group ? Look at our parent's sibling
+                var tbody = $cell.closest('tbody');
+                if (!tbody.length) {
+                    // We can have th in tbody so we can't simply look for thead
+                    // if cell is a th and tbody instead
+                    tbody = $cell.closest('thead');
+                }
+
+                var connectedGroup = findConnected(tbody, 'tbody');
+                if (!connectedGroup.length) {
+                    // Maybe the next group is a thead ?
+                    connectedGroup = findConnected(tbody, 'thead');
+                }
+                if (connectedGroup.length) {
+                    // Found another group
+                    // TODO: focus correct row
+                    var connectedRows = connectedGroup.find('tr');
+                    var rowIndex;
+                    if (direction === 'prev') {
+                        rowIndex = connectedRows.length - 1
+                    } else {
+                        rowIndex = 0;
+                    }
+                    connectedRow = connectedRows.eq(rowIndex);
+                } else {
+                    // End of the table
+                    return;
+                }
+            }
+
+            var connectedCell;
+            if (connectedRow.hasClass('o_group_header')) {
+                connectedCell = connectedRow.children();
+                connectedCell.data('o_list_col_index', colIndex);
+            } else if (connectedRow.has('td.o_group_field_row_add').length) {
+                connectedCell = connectedRow.find('.o_group_field_row_add');
+                connectedCell.data('o_list_col_index', colIndex);
```

Comment:
try to use this.currentFieldIndex or create a new one instead rather than writing it on the cells like that